django/urls/resolvers.py

```diff
@@ -351,7 +351,8 @@ def regex(self):
     @property
     def language_prefix(self):
         language_code = get_language() or settings.LANGUAGE_CODE
-        if language_code == settings.LANGUAGE_CODE and not self.prefix_default_language:
+        default_language = get_supported_language_variant(settings.LANGUAGE_CODE)
+        if language_code == default_language and not self.prefix_default_language:
```

Comment:
This change is due to the fact that we're currently activating a fallback language not `settings.LANGUAGE_CODE`:

https://github.com/django/django/blob/5b8a043bf51ab8bcf4a758d0b4646f30a84be183/django/middleware/locale.py#L40-L41

@claudep What do you think?

Reply:
I wanted to emphasize that currently we're activating a fallback language e.g. `en`. In older versions, we activated `settings.LANGUAGE_CODE`, e.g. `en-us`. It is a side-effect of 94e7f471c4edef845a4fe5e3160132997b4cca81, not an intended change (as far as I'm aware). Now the question :smile: Do you think it's fine? or maybe we should try to avoid this side-effect instead of changing the `language_prefix()` :thinking: .

Reply:
The test would also pass by adding `en-us` to the global `LANGUAGES` setting. `en-us` is the default for `LANGUAGE_CODE` but it doesn't appear to be in any of the lists of supported languages (and it falls back to `en`)

Reply:
I think updating the default for `LANGUAGE_CODE` to `en` makes sense.

I guess previously we didn't validate that the `LANGUAGE_CODE` setting was within `LANGUAGES`, that `en-us` is the default is unfortunate but I guess people could also have used other things. :thinking: 

Some options could be:
- Go with this current change and anyone who has this invalid value for `LANGUAGE_CODE` would not need to make an update. 
- We could treat `en-us` as a alias for `en` (so similar to this change but a bit stricter to only handle the `en-us` case).
-  Tell people to update the setting. 

I'm really not familiar with depreciating things, maybe we should add a warning when your `LANGUAGE_CODE` is not within `LANGUAGES`?

Reply:
Thanks both, let's have it :+1: 

> If we keep that, we should also probably change the default LANGUAGE_CODE to en in global settings. Does that make sense?

Yes, but we can do this separately as a cleanup in Django 5.0.